swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/events/ResourceStateManager.kt

```diff
@@ -171,28 +172,51 @@ class ResourceStateManager(
     resource: MarkedResource,
     workConfiguration: WorkConfiguration
   ): String {
-    val taskId = taggingService.upsertImageTag(
-      UpsertImageTagsRequest(
-        imageNames = setOf(resource.name ?: resource.resourceId),
+    val taskId = with(resource.resourceType) {
+      when {
+        equals("serverGroup", true) -> tagAsg(resource)
+        equals("image", true) -> tagImage(resource)
+        else -> log.error("Failed to tag resource ${resource.uniqueId()}")
```

Comment:
When this is the case (not server group or image) `taskId` will be null and then it will NPE below, right?

Also, what happens in the case of a snapshot?

Reply:
good point , yeah we don't have an option to optout today for any other resource types today, do you think it would be better to handle this by throwing an exception instead ?

Reply:
We should have the option to opt out for snapshot - and I thought that was working.

If we correctly opt out the resource but just don't know how to tag it then I don't think we should throw an exception here, but we will need to handle a `null` `taskId` below.

Reply:
Updated the code for this. You are right `snapshot` outputs work fine , i was referring to the `aws` `expiration_time` tagging.